backend/models/scene.js

```diff
@@ -16,53 +16,56 @@
  */
 
 "use strict";
-const mongoose = require("mongoose");
-const ModelFactory = require("./factory/modelFactory");
-const Schema = mongoose.Schema;
 const utils = require("../utils");
 const db = require("../handler/db");
 const ExternalServices = require("../handler/externalServices");
 const matrix = require("./helper/matrix");
 
-const schema = Schema({
-	_id: Object,
-	parents: []
-});
+function cleanOne(sceneToClean) {
+	if (sceneToClean) {
+		if (sceneToClean._id) {
+			sceneToClean._id = utils.uuidToString(sceneToClean._id);
+		}
+
+		if (sceneToClean.parents) {
+			sceneToClean.parents = sceneToClean.parents.map(p => utils.uuidToString(p));
+		}
+	}
 
-if (!schema.options.toJSON) {
-	schema.options.toJSON = {};
+	return sceneToClean;
 }
 
-const dbFindOne = async (account, model, query, projection) => {
-	const coll = await db.getCollection(account, model + ".scene");
-	return await coll.findOne(query, projection);
-};
+function clean(scenesToClean) {
+	return scenesToClean.map(cleanOne);
+}
 
-schema.options.toJSON.transform = function (doc, ret) {
-	ret._id = utils.uuidToString(doc._id);
-	if(doc.parents) {
-		const newParents = [];
-		doc.parents.forEach(function(parentId) {
-			newParents.push(utils.uuidToString(parentId));
-		});
-		ret.parents = newParents;
-	}
-	return ret;
+function getSceneCollectionName(model) {
+	return model + ".scene";
+}
+
+const Scene = {};
+
+Scene.findOneScene = async function (account, model, query, projection) {
+	return cleanOne(await db.findOne(account, getSceneCollectionName(model), query, projection));
 };
 
-schema.statics.getBySharedId = async (account, model, shared_id, revisionIds, projection = {}) => {
-	return await dbFindOne(account, model, {shared_id, _id :{$in: revisionIds}}, projection);
+Scene.findScenes = async function (account, model, query, projection) {
+	return clean(await db.find(account, getSceneCollectionName(model), query, projection));
 };
 
-schema.statics.getObjectById = async (account, model, id, projection = {}) => {
-	return await dbFindOne(account, model, {_id: id}, projection);
+Scene.getBySharedId = async function (account, model, shared_id, revisionIds, projection = {}) {
```

Comment:
for consistency, should call this `getNodeBySharedId`

Reply:
@carmenfan I'm also removing it from Scene so it's not exported outside as nothing else seems to use it